src/cpp/src/visual_language/qwen2_5_vl/classes.cpp

```diff
@@ -14,89 +14,65 @@ std::pair<ov::Tensor, std::vector<int32_t>> get_window_index(
     const ProcessorConfig& processor_config,
     const VLMConfig& vlm_config
 ) {
-    std::vector<size_t> window_indices;
+    std::vector<int64_t> window_indices;
     std::vector<int32_t> cu_window_seqlens = {0};
-    size_t window_index_id = 0;
-    
+    int32_t window_index_id = 0;
+
     const size_t spatial_merge_size = processor_config.merge_size;
     const size_t spatial_merge_unit = spatial_merge_size * spatial_merge_size;
     const size_t vit_merger_window_size = vlm_config.vision_config_window_size / spatial_merge_size / vlm_config.vision_config_patch_size;
-    
+
     for (const auto& grid_thw : grids_thw) {
         size_t grid_t = grid_thw.at(0);
         size_t grid_h = grid_thw.at(1);
         size_t grid_w = grid_thw.at(2);
-        
-        // Calculate merged grid dimensions
+
         size_t llm_grid_h = grid_h / spatial_merge_size;
         size_t llm_grid_w = grid_w / spatial_merge_size;
-        
-        // Calculate padding for making dimensions divisible by window size
+
         size_t pad_h = (vit_merger_window_size - llm_grid_h % vit_merger_window_size) % vit_merger_window_size;
         size_t pad_w = (vit_merger_window_size - llm_grid_w % vit_merger_window_size) % vit_merger_window_size;
-        
+
         size_t num_windows_h = (llm_grid_h + pad_h) / vit_merger_window_size;
         size_t num_windows_w = (llm_grid_w + pad_w) / vit_merger_window_size;
-        
-        // Create and fill indices matrix with sequential indices
-        std::vector<std::vector<std::vector<int32_t>>> index_3d(grid_t, 
-            std::vector<std::vector<int32_t>>(llm_grid_h, 
-                std::vector<int32_t>(llm_grid_w, 0)));
-        
-        size_t index = 0;
+
+        std::vector<std::vector<std::vector<int32_t>>> index_3d(grid_t,
+            std::vector<std::vector<int32_t>>(llm_grid_h + pad_h,
+                std::vector<int32_t>(llm_grid_w + pad_w, -100)));
```

Comment:
quite heavy construction of nested vectors. I think we can re-implement this part as well to have 1D vector.

Reply:
ok. Let me do it